### PR TITLE
Implement Parent Rosters in Group Toolbox for Children's Small Groups #2584

### PR DESCRIPTION
**Pull request by dcs619**
## Contributor Agreement
_By contributing your code, you agree to license your contribution under the [Rock Community License Agreement](https://www.rockrms.com/license)._
✅

## Context
_What is the problem you encountered that lead to you creating this pull request?_

This PR adds Parent Roster support for Children's Small Groups. This allows leaders to have access to parental information, not just information about the child they lead.  See discussion, explanation and screenshots at #2584.

## Possible Implications
_What could this change potentially impact? Are there any security considerations? Where could this potentially affect backwards compatibility?_

If this is merged to v8, @edmistj suggested using the Person.AgeClassification field to target Groups instead of depending on the GroupMember.GroupRole of Child in family groups.

## Migrations
Should your pull request require a migration, please exclude the migration from the Rock.Migration project, but submit it in your pull request. Please add a note to your pull request that provides a heads up that a migration file is present.

The block setting for Enable Parent Support is off by default so this does not include a migration to add a new "Parent Roster Tab" page beneath the current Group Toolbox page.  If that's something I need to do I can include it.

### CODE REVIEW

**Comment by edmistj** [comment]
@dcs619 How did you decide to determine if someone is a child?

**Comment by dcs619** [comment]
@edmistj When [GetGroupMemberPeople](https://github.com/SparkDevNetwork/Rock/pull/2855/files#diff-c5a03cc4dc3377f85c43183b3b8a2bd8R1138) is called with transformParent = true, it checks if a Person in the Group has a Family.ChildRole, and returns any People with the Family.AdultRole.  This is the same code used by Person.GetFamilies(), but called once for all group members instead of per person.

Short version: if a Person doesn't have a Parent, they can't be a Child.  I think Rock allows a Family with a single Child (no Adult), but in that case they don't need a row on the Parent Roster.

**Comment by azturner** [comment]
@dcs619 Were you going to update this to use the AgeClassification (see discussion on #2854)

**Comment by dcs619** [comment]
@azturner Yes, sorry I missed the notification on that discussion
